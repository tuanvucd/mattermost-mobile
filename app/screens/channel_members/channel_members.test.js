// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';

import Preferences from 'mattermost-redux/constants/preferences';

import {shallowWithIntl} from 'test/intl-test-helper';

import CustomList from 'app/components/custom_list';
import ChannelMembers from './channel_members';

describe('ChannelMembers', () => {
    const baseProps = {
        theme: Preferences.THEMES.default,
        currentUserId: 'current-user-id',
        currentChannelId: 'current-channel-id',
        canManageUsers: false,
        actions: {
            getProfilesInChannel: jest.fn().mockImplementation(() => Promise.resolve()),
            handleRemoveChannelMembers: jest.fn(),
            searchProfiles: jest.fn(),
            setButtons: jest.fn(),
            popTopScreen: jest.fn(),
        },
        componentId: 'component-id',
        isLandscape: false,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <ChannelMembers {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should use renderUnselectableItem when canManagerUsers is false', () => {
        const props = {...baseProps, canManageUsers: false};

        const wrapper = shallowWithIntl(
            <ChannelMembers {...props}/>,
        );

        const renderItem = wrapper.find(CustomList).props().renderItem;
        expect(renderItem).toEqual(wrapper.instance().renderUnselectableItem);
    });

    test('should use renderSelectableItem when canManagerUsers is true', () => {
        const props = {...baseProps, canManageUsers: true};

        const wrapper = shallowWithIntl(
            <ChannelMembers {...props}/>,
        );

        const renderItem = wrapper.find(CustomList).props().renderItem;
        expect(renderItem).toEqual(wrapper.instance().renderSelectableItem);
    });
});