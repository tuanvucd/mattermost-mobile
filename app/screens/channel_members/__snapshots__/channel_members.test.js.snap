// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelMembers should match snapshot 1`] = `
<ForwardRef(forwardConnectRef)>
  <Connect(StatusBar) />
  <View
    style={
      Array [
        Object {
          "marginVertical": 5,
        },
        null,
      ]
    }
  >
    <SearchBarIos
      autoCapitalize="none"
      backgroundColor="transparent"
      blurOnSubmit={true}
      cancelTitle="Cancel"
      inputHeight={33}
      inputStyle={
        Object {
          "backgroundColor": "rgba(61,60,64,0.2)",
          "color": "#3d3c40",
          "fontSize": 15,
        }
      }
      keyboardAppearance="light"
      leftComponent={null}
      onBlur={[Function]}
      onCancelButtonPress={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSearchButtonPress={[Function]}
      onSelectionChange={[Function]}
      placeholder="Search"
      placeholderTextColor="rgba(61,60,64,0.5)"
      searchIconCollapsedMargin={10}
      searchIconExpandedMargin={10}
      tintColorDelete="rgba(61,60,64,0.5)"
      tintColorSearch="rgba(61,60,64,0.5)"
      titleCancelColor="#3d3c40"
      value=""
    />
  </View>
  <CustomList
    data={Array []}
    extraData={Object {}}
    isLandscape={false}
    listType="section"
    loading={false}
    loadingComponent={null}
    noResults={null}
    onLoadMore={[Function]}
    onRowPress={[Function]}
    renderItem={[Function]}
    shouldRenderSeparator={true}
    showNoResults={true}
    theme={
      Object {
        "awayIndicator": "#ffbc42",
        "buttonBg": "#166de0",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3d3c40",
        "codeTheme": "github",
        "dndIndicator": "#f74343",
        "errorTextColor": "#fd5960",
        "linkColor": "#2389d7",
        "mentionBj": "#ffffff",
        "mentionColor": "#145dbf",
        "mentionHighlightBg": "#ffe577",
        "mentionHighlightLink": "#166de0",
        "newMessageSeparator": "#ff8800",
        "onlineIndicator": "#06d6a0",
        "sidebarBg": "#145dbf",
        "sidebarHeaderBg": "#1153ab",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#579eff",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#4578bf",
        "sidebarUnreadText": "#ffffff",
        "type": "Mattermost",
      }
    }
  />
</ForwardRef(forwardConnectRef)>
`;
